ci_tests/src/index.ts

```diff
@@ -134,215 +205,327 @@ async function deployEth(
 
   // Setup the initial calls, like transceivers for the manager
   console.log("Set transceiver for manager");
-  await (await manager.setTransceiver(await transceiver.address)).wait();
+  await tryAndWaitThrice(() => manager.setTransceiver(transceiver.address));
 
   console.log("Set outbound limit for manager");
-  await (await manager.setOutboundLimit(utils.parseEther("10000"))).wait();
+  await tryAndWaitThrice(() =>
+    manager.setOutboundLimit(utils.parseEther("10000"))
+  );
 
   return {
+    type: "evm",
     chainId: chain_id,
-    transceiverAddress: await transceiverProxyAddress.address,
-    managerAddress: await managerProxyAddress.address,
+    chainName,
+    transceiverAddress: transceiverProxyAddress.address,
+    managerAddress: managerProxyAddress.address,
     NTTTokenAddress: ERC20NTTAddress,
     wormholeCoreAddress: "0xC89Ce4735882C9F0f0FE26686c53074E09B0D550", // Same on both of the chains
-    rpcEndpoint: rpc_endpoint,
+    signer,
   };
 }
 
-async function link(chain1: ChainDetails, chain2: ChainDetails) {
+async function initSolana(): Promise<SolanaChainDetails> {
+  console.log("Using public key", SOL_PUBLIC_KEY.toString());
+  const mint = await spl.createMint(
+    SOL_CONNECTION,
+    SOL_PRIVATE_KEY,
+    SOL_PUBLIC_KEY,
+    null,
+    9
+  );
+  console.log("Created mint", mint.toString());
+  const tokenAccount = await spl.createAssociatedTokenAccount(
+    SOL_CONNECTION,
+    SOL_PRIVATE_KEY,
+    mint,
+    SOL_PUBLIC_KEY
+  );
+  console.log("Created token account", tokenAccount.toString());
+  // await spl.mintTo(
+  //   SOL_CONNECTION,
+  //   SOL_PRIVATE_KEY,
+  //   mint,
+  //   tokenAccount,
+  //   SOL_PRIVATE_KEY,
+  //   BigInt(10000000)
+  // );
+  // console.log("Minted 10000000 tokens");
+  await spl.setAuthority(
+    SOL_CONNECTION,
+    SOL_PRIVATE_KEY,
+    mint,
+    SOL_PRIVATE_KEY,
+    0, // mint
+    SOL_NTT_CONTRACT.tokenAuthorityAddress()
+  );
+  console.log(
+    "Set token authority to",
+    SOL_NTT_CONTRACT.tokenAuthorityAddress().toString()
+  );
+
+  await SOL_NTT_CONTRACT.initialize({
+    payer: SOL_PRIVATE_KEY,
+    owner: SOL_PRIVATE_KEY,
+    chain: "solana",
+    mint,
+    outboundLimit: new BN(1000000000),
+    mode: "burning",
+  });
+  console.log(
+    "Initialized ntt at",
+    SOL_NTT_CONTRACT.program.programId.toString()
+  );
+
+  await SOL_NTT_CONTRACT.registerTransceiver({
+    payer: SOL_PRIVATE_KEY,
+    owner: SOL_PRIVATE_KEY,
+    transceiver: SOL_NTT_CONTRACT.program.programId,
+  });
+  console.log("Registered transceiver with self");
+
+  return {
+    type: "solana",
+    chainId: 1,
+    chainName: "solana",
+    transceiverAddress: SOL_NTT_CONTRACT.emitterAccountAddress().toString(),
+    managerAddress: SOL_NTT_CONTRACT.program.programId.toString(),
+    NTTTokenAddress: mint.toString(),
+    wormholeCoreAddress: SOL_CORE_ADDRESS,
+    signer: SOL_PRIVATE_KEY,
+  };
+}
+
+async function link(
```

Comment:
I was trying to find a more general way to call these functions, since we may be adding more NTT chains in the future. 

Right now, it's fine but if we support 7 stacks this will become awful to do the peer registration for all of them together. 

Don't have to change right now but something to think about.

Reply:
agree, I'd like to address in the next PR :pray: